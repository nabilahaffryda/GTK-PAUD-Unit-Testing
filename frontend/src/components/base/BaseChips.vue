<template>
  <v-chip class="ma-1 pr-4" :color="color ? color : M_WARNA[colorIdx || 0]" small>
    <v-avatar v-if="chip.icon">
      <v-icon small>{{ chip.icon }}</v-icon>
    </v-avatar>
    {{ chip.label }}
  </v-chip>
</template>

<script>
export default {
  name: 'Chips',
  props: {
    chip: {
      type: Object,
      required: true,
    },
    color: {
      type: String,
    },
    colorIdx: {
      type: Number,
    },
  },
  data() {
    return {
      M_WARNA: {
        0: 'error-box',
        1: 'success-box',
        2: 'warning-box',
        3: 'primary-box',
        4: 'second-box',
      },
    };
  },
};
</script>

<style scoped>
/* v-chip coloring */
.success-box {
  background-color: #bbf5bb !important;
  border-color: #99c995 !important;
  color: #19601c !important;
}
.warning-box {
  background-color: #ffeaa7 !important;
  border-color: #d9c483 !important;
  color: #8a6a2f !important;
}
.error-box {
  background-color: #ffcdd2 !important;
  border-color: #cb9ca1 !important;
  color: #7f0000 !important;
}
.primary-box {
  background-color: #e3f2fd !important;
  border-color: #bbdefb !important;
  color: #2196f3 !important;
}
.second-box {
  background-color: #fafafa !important;
  border-color: #f5f5f5 !important;
  color: #9e9e9e !important;
}
</style>
