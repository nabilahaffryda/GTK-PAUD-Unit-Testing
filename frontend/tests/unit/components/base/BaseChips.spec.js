import { mount, createLocalVue } from "@vue/test-utils";
import BaseChips from "@/components/base/BaseChips.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';

Vue.use(Vuetify)
const localVue = createLocalVue();

describe('BaseChips.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })

    test('should show chip', () => {
        const VChip = {
            props: ['color', 'colorIdx', 'chip'],
            template: '<div><slot :chip="chip"/></div>'
        }
        const wrapper = mount(BaseChips, {
            localVue,
            vuetify,
            stubs: {
                VChip
            },
            propsData: {
                chip: {}
            }
        });
        expect(wrapper).toMatchSnapshot();
    })
})
