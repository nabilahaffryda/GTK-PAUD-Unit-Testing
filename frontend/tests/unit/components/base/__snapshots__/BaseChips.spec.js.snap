// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseChips.vue should show chip 1`] = `
<div small="" class="ma-1 pr-4">
  <!---->

</div>
`;
